app/main.py

```diff
@@ -1,2 +1,47 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.storage = [None] * self.capacity
+
+    class Node:
+        def __init__(self, key: Hashable, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.next = None
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = hash(key) % self.capacity
+        node = self.storage[index]
+
+        while node:
+            if node.key == key:
+                node.value = value
+                return
+            node = node.next
+
+        new_node = self.Node(key, value)
+        new_node.next = self.storage[index]
+        self.storage[index] = new_node
+        self.size += 1
```

Comment:
Consider implementing a resizing mechanism to handle cases where the dictionary becomes too full. This will help maintain efficient operations as more items are added. Typically, resizing is done when the load factor exceeds a certain threshold (e.g., 0.7).